src/services/document-service.spec.js

```diff
@@ -312,4 +312,70 @@ describe('document-service', () => {
 
   });
 
+  describe('whenSearchingByTags', () => {
+    let doc1 = null;
+    let doc2 = null;
+    let doc3 = null;
+    beforeEach(async () => {
+      doc1 = await createTestDocument(container, user, {
+        title: 'Doc 1',
+        slug: 'doc-1',
+        sections: [],
+        tags: ['music', 'instructor', 'Dj', 'Cretu']
+      });
+
+      doc2 = await createTestDocument(container, user, {
+        title: 'Doc 2',
+        slug: 'doc-2',
+        sections: [],
+        tags: ['Music', 'Instructor', 'Goga']
+      });
+
+      doc3 = await createTestDocument(container, user, {
+        title: 'Doc 3',
+        slug: 'doc-3',
+        sections: [],
+        tags: ['Wolf', 'gang', 'from', 'Beat', 'oven', 'music']
+      });
+    });
+
+    describe('when I search for something that should not match', () => {
+      it('should return an empty array', async () => {
+        const results = await sut.getDocumentsByTags('I can not find anything in this db');
+        expect(results).toHaveLength(0);
+      });
+    });
+
+    describe('when I search with a query that returns a single document', () => {
+      it('should project the data correctly', async () => {
+        const results = await sut.getDocumentsByTags('Wolf gang beat Oven');
+
+        expect(results).toHaveLength(1);
+        const result = results[0];
+        expect(result.title).toEqual(doc3.title);
+        expect(result.slug).toEqual(doc3.slug);
+        expect(result.tags).toEqual(doc3.tags);
+        expect(result.tagMatchCount).toEqual(4);
+        expect(result.updatedOn).not.toBeNull();
+        expect(result.sections).toBeUndefined();
+      });
+    });
+
+    describe('when I search with a query that returns multiple documents', () => {
+      it('contain all documents with the correect tag match count', async () => {
+        const results = await sut.getDocumentsByTags('music instructor goga');
+
+        expect(results).toHaveLength(3);
+
+        const resultMap = results.reduce((acc, doc) => {
+          acc[doc.title] = { ...doc };
+          return acc;
+        }, {});
+
+        expect(resultMap[doc1.title].tagMatchCount).toEqual(2);
+        expect(resultMap[doc2.title].tagMatchCount).toEqual(3);
+        expect(resultMap[doc3.title].tagMatchCount).toEqual(1);
+      });
```

Comment:
Maybe you could also add a test for this unhappy path.
https://github.com/educandu/educandu/pull/9/files#r745483179

Actually might be nice to have a `const noSearchResultCases = [..]` data driven test and also include searching by string with spaces, tags that are all shorter than 3 or longer than 30..

Reply:
This should be covered by the test with a tab and sequential spaces.